### PR TITLE
test: add test for mocha4

### PR DESCRIPTION
**Pull request by popomore**
mocha4 has released https://boneskull.com/mocha-v4-nears-release

and node pre-4.0.0 has dropped

### CODE REVIEW

**Comment by popomore** [comment]
@blakeembrey

**Comment by blakeembrey** [comment]
Wouldn’t the change you did to before_script mean it’s testing the same mocha version everywhere now?

**Comment by popomore** [comment]
It's better to install mocha before `npm install`, but `before_script` will run after `npm install`, so I move the script to `before_install`.

It's testing every version that specified in env, you can see the log in travis.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "build": "browserify lib/co-mocha.js -s co-mocha -o co-mocha.js",
     "test-spec": "mocha --harmony -R spec --require lib/co-mocha.js --bail",
     "test-cov": "testem ci -l Firefox,Node",
-    "test": "npm run lint && npm run build && npm run test-cov"
+    "test": "mocha -V && npm run lint && npm run build && npm run test-cov"
```
Comment (popomore) [inline package.json]: I have printed the version

**Comment by popomore** [comment]
@blakeembrey any other comment?

**Comment by popomore** [comment]
ping @blakeembrey

**Pull request merged**